fs_storage/models/fs_storage.py

```diff
@@ -339,6 +339,21 @@ def _get_fs_options(self):
             return options
         return self._eval_options_from_env(self.json_options)
 
+    def _get_marker_file_name(self):
+        return ".odoo_fs_storage_%s.marker" % self.id
+
+    def _check_connection(self, fs=False):
```

Comment:
I have some doubts: this `_check_connection` method is not called by `def fs` property, so the FS object returned by `_get_filesystem` will continue to persist in the worker memory, even if its socket gets closed over time.
Should we not call this method from `def fs` as well?

What we did to avoid such connection errors is a quick&dirty patch to always return a new FS object through `def fs` property, this adds some overhead but guarantee the connection has been refreshed.

Reply:
@JordiMForgeFlow it should be fine like this from what I see in my tests :+1: seems the property is only useful within the same Odoo transaction, the `self.__fs` object doesn't persist in memory between 2 HTTP requests on Odoo (`__init__` is always called and resets `self.__fs` to None, so we always enter in the `if not self.__fs:` block in the property), so all good.